Setting user's password

do-kevin opened the pull request.

do-kevin committed:
Adds verifyEmail POST req

Removes unused styles

Adds missing trailing comma

Init PR

Adds eslint and peerDependencies

Init ESLint config

Applies ESLint

Adds and uses babel-eslint

Updates App.js with eslint

Renames file for better clarity

do-kevin committed:
Adds setPassword and resetPassword post req funcs

do-kevin committed:
Checks for verification, redirects

do-kevin committed:
Splits LoginRegistration to import two components

Init LoginForm w/ switch form btn

Adds a btn to allow switching to RegisterForm

Adds login post req for LoginForm.js

Init CreateAccount

Fixes Redirect and Route to CreateAccount after verification

Adds verifyEmail POST req

Removes unused styles

Adds missing trailing comma

Adds PropTypes for ESLint

do-kevin committed:
Gives match params for CreateAccount

Edits margin bottom

do-kevin committed:
Sets up password resetting

Updates setPassword func with axios config

Shorter cooldown for redirect

Adds defaultConfigs and axiosConfigs for Axios

Early password form with no matching fields or validation yet

Fixes headers typo

Adds password validation

do-kevin committed:
Sets up password resetting

Shorter cooldown for redirect

Fixes all headers typo

Uses qs.stringify for verifyEmail due to x-www-form-urlencoded

Adds conditional rendering for statusText

Replaces inputs w/ Material UI TextFields

Removes unused input and switch styles

Adds bluebird depdendency

Updates Verification screen w/ Promise.delays and Circular Progress

Refactors setState()

Init React Context

Replaces Grid and Paper w/ CenterPaper

Applies early React slots

Removes redundant MUI styles

Puts in CenterPaper and tweaks

Replaces with InputForms

Adds new prop

Adds feedback text

Removes classnames and adds inflection

lan-nguyen91 commented on src/Screens/ConfirmPassword.js:
```diff
@@ -0,0 +1,124 @@
+                  </div>
+                  <Grid item>
+                    <Button
+                      className={classNames(classes.CustomBtn)}
```
no need classNames

lan-nguyen91 commented on src/Screens/LoginRegistration.js:
```diff
@@ -0,0 +1,71 @@
+    const { showLogin, showResetPassword } = this.state;
+    let renderForm;
+
+    if (showLogin === true && showResetPassword === false) {
```
`if (showLogin && !showResetPassword)`

lan-nguyen91 commented on src/Screens/LoginRegistration.js:
```diff
@@ -0,0 +1,71 @@
+class LoginRegistration extends Component {
+  state = {
+    showLogin: true,
+    showResetPassword: false,
```
showLogin: boolean,
showReset: boolean,
showRegistration: boolean,

toggle(formType) => {
     let obj = Object.assign({},  { showLogin: false, showRegistration: false, showReset: false }, {
            [formType]: true,
     })

      this.setState(obj)
}

lan-nguyen91 commented on src/Screens/LoginForm.js:
```diff
@@ -0,0 +1,145 @@
+            <Formik
+              initialValues={{ email: '', password: '' }}
+              onSubmit={values => userLogin(values.email, values.password)}
+              validate={({ email, password }) => {
```
use validation schema
https://jaredpalmer.com/formik/docs/api/errormessage

lan-nguyen91 commented on src/Screens/LoginForm.js:
```diff
@@ -0,0 +1,145 @@
+        className={classes.root}
+      >
+        <Grid item>
+          <Paper className={classes.paper}>
```
<>

lan-nguyen91 commented on src/Screens/LoginRegistration.js:
```diff
@@ -0,0 +1,71 @@
+
+
+    return (
+      <div>
```
div => CenterCard

lan-nguyen91 commented on src/Screens/Verification.js:
```diff
@@ -0,0 +1,107 @@
+    profileId: '',
+  };
+
+  componentDidMount() {
```
async

lan-nguyen91 commented on src/Screens/Verification.js:
```diff
@@ -0,0 +1,107 @@
+    const { token, user_id } = parsed;
+
+    console.log(parsed);
+    verifyEmail(user_id, token)
```
await

lan-nguyen91 commented on src/Screens/Verification.js:
```diff
@@ -0,0 +1,107 @@
+          this.setState({
+            isRequesting: false,
+            profileToken: userToken,
+            profileId: user_id,
```
isVerified should be in here ?

lan-nguyen91 commented on src/Screens/Verification.js:
```diff
@@ -0,0 +1,107 @@
+            isRedirecting: true,
+          }), 10000));
+        }
+
```
use `bluebird`
import Promise from 'bluebird';

await Promise.delay(1000);

lan-nguyen91 replied:
setState redirect;

do-kevin committed:
Implements React Context

Adds react-redux

Init Loading component

do-kevin committed:
Adds dynamic importing

Adds Yup validation

lan-nguyen91 commented on .eslintrc:
```diff
@@ -0,0 +1,34 @@
+				"allowChildren": false
+			}
+		]
+	}
```
"react/prop-types": "off"

lan-nguyen91 commented on src/Services/RegistrationAPI.js:
```diff
@@ -1,27 +1,113 @@
+/* eslint-disable no-console */
 import axios from 'axios';
+import queryString from 'query-string';
 
```
const instance = axios.create( baasaurl)

lan-nguyen91 commented on src/Services/RegistrationAPI.js:
```diff
@@ -1,27 +1,113 @@
+  },
+};
+
+async function registerEmail(userEmail = '') {
```
try to use NPprogress

lan-nguyen91 commented on src/Services/RegistrationAPI.js:
```diff
@@ -1,27 +1,113 @@
+      email: userEmail,
+    }, defaultConfig);
+    return response;
+  } catch (error) {
```
try to use growl, prefer material ui snack-bar

lan-nguyen91 commented on src/Screens/Verification.js:
```diff
@@ -1,24 +1,124 @@
+      this.setState({
+        isRequesting: false,
+      });
+      Promise.delay(4000).then(() => {
```
await

lan-nguyen91 replied:
await Promise.delay();
setState
await Promise.delay();
setState

do-kevin committed:
Sets up Redux w/ example too

Ignores proptypes rule

Adds await to Promises.delays

do-kevin committed:
Adds multi-nprogress

Some returns show up in a MUI snack bar

Adds disabled prop for btn to prevent over-submitting like resetting password

do-kevin committed:
Returns errors in snack bars
